core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -50,6 +58,35 @@ object ZkUtils extends Logging {
   val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
 
+  /** true if java.security.auth.login.config is set to some jaas file which has "Client" entry. **/
+
+  val isSecure: Boolean = {
+    val loginConfigurationFile: String = System.getProperty("java.security.auth.login.config")
+    var isSecurityEnabled = false
+    if (loginConfigurationFile != null && loginConfigurationFile.length > 0) {
+      val configFile: File = new File(loginConfigurationFile)
+      if (!configFile.canRead) {
+        throw new KafkaException(s"File $loginConfigurationFile cannot be read.")
+      }
+      try {
+        val configUri: URI = configFile.toURI
+        val loginConf = Configuration.getInstance("JavaLoginConfig", new URIParameter(configUri))
+        isSecurityEnabled = loginConf.getAppConfigurationEntry("Client") != null
```

Comment:
When we add SASL for the broker, we need to pass in the JAAS login config for the broker, which can be different from that for ZK. So, instead of using "Client" as the name, perhaps, we should name it "ZookeeperClient" to make it clear?